.github/workflows/create_llvm.yml

```diff
@@ -0,0 +1,78 @@
+name: cache_llvm
+
+on:
+  push:
+    branches:
+      - main
+    path: .github/workflows/create_llvm.yml
+  workflow_dispatch:
+
+jobs:
+  create_llvm_cache:
+    strategy:
+      matrix:
+        version: [15, 16]
+        os: [ubuntu-22.04]
+        build_type: [Release, RelAssert]
+        include:
+          # os_flags and build_type_flags are new and added to each matrix entry and therefore don't create a new one
+          # Note that We need to use RelAssert as the cache key matching can match Release against ReleaseAssert
+          - os_flags:
+          - build_type_flags:
```

Comment:
the rules are kind of complicated.
If you add variables here, it doesn't create a new matrix dimension, it just sets the variables - basically we need a default, otherwise it effectively creates an extra matrix run.

Reply:
I see, thanks